pmlc/target/x86/stencil_tpp_unary.cc

```diff
@@ -209,34 +271,82 @@ class StencilImpl : public pxa::StencilBase {
 
 public:
   explicit StencilImpl(AffineParallelOp op)
-      : StencilBase(
-            op,
-            {
-                pxa::StencilIndexRequirement{
-                    /*idxName=*/"eltwise_i",
-                    /*tilingGenerator=*/pxa::ExactRangeGenerator(),
-                    pxa::IndexStridePredicates{
-                        [](int64_t stride) { return stride > 1; }, // output
-                        [](int64_t stride) {
-                          return (stride == 0 || stride > 1);
-                        }, // input
-                    }},
-                pxa::StencilIndexRequirement{
-                    /*idxName=*/"eltwise_j",
-                    /*tilingGenerator=*/pxa::ExactRangeGenerator(),
-                    pxa::IndexStridePredicates{
-                        [](int64_t stride) { return stride == 1; }, // output
-                        [](int64_t stride) {
-                          return (stride == 1 || stride == 0);
-                        }, // input
-                    }},
-            }) {}
+      : StencilBase(op, {
+                            pxa::StencilIndexRequirement{
+                                /*idxName=*/"eltwise_i",
+                                /*tilingGenerator=*/pxa::ExactRangeGenerator(),
+                                pxa::IndexStridePredicates{
+                                    [](int64_t stride) {
+                                      return (stride == 0 || stride > 1);
+                                    }, // output
+                                    [](int64_t stride) {
+                                      return (stride == 0 || stride > 1);
+                                    }, // input
+                                }},
+                            pxa::StencilIndexRequirement{
+                                /*idxName=*/"eltwise_j",
+                                /*tilingGenerator=*/pxa::ExactRangeGenerator(),
+                                pxa::IndexStridePredicates{
+                                    [](int64_t stride) {
+                                      return (stride == 1 || stride == 0);
+                                    }, // output
+                                    [](int64_t stride) {
+                                      return (stride == 1 || stride == 0);
+                                    }, // input
+                                }},
+                        }) {}
 };
 
 } // namespace
 
+// For 1D stenciling, we still map to 2D TPPs. We can make 2D TPPs do 1D
+// by having the outer dim being 1.
+void addSingleIteration(AffineParallelOp op) {
+  if (op.getIVs().size() == 1) {
```

Comment:
Can you have an early exit instead of wrapping everything in a 1d loop? Can you please add documentation for the method?